### PR TITLE
gasket: init at 1.0-18

### PR DESCRIPTION
**Pull request by kylehendricks**
###### Description of changes
Added this driver as it's needed to use the [mini-pcie Google Coral EdgeTPU](https://coral.ai/products/pcie-accelerator).

The Google Cora EdgeTPU is commonly used for object detection in security camera footage using [Frigate](https://github.com/blakeblackshear/frigate).

https://github.com/google/gasket-driver

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
  - Ensured module loaded.
  - Tested with a google coral edgeTPU using [Frigate](https://github.com/blakeblackshear/frigate).
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -8344,6 +8344,13 @@
     githubId = 2422454;
     name = "Kai Wohlfahrt";
   };
+  kylehendricks = {
+    name = "Kyle Hendricks";
+    email = "kyle-github@mail.hendricks.nu";
+
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:8350]:
```suggestion
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -26204,6 +26204,10 @@ with pkgs;
 
   fwts = callPackage ../os-specific/linux/fwts { };
 
+  gasket = callPackage ../os-specific/linux/gasket {
+    inherit (linuxPackages) kernel;
+  };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:26209]: Shouldn't this then go to os-specific packages?

Comment (kylehendricks) [reply pkgs/top-level/all-packages.nix:26209]: I wasn't 100% sure where I was supposed to put this.  Is this supposed to go into a different file like `linux-kernels.nix`?

Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix:26209]: Oh, wait, it is actually there already.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (SuperSandro2000) [reply maintainers/maintainer-list.nix:8350]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

**Pull request merged**